### PR TITLE
Add short

### PR DESCRIPTION
**Pull request by EitanSomething**
This currently breaks all ofx modifiers and I can't seem to figure out why. @eliemichel can you take a look at it.

### CODE REVIEW

**Comment by eliemichel** [comment]
Mmh maybe you mixed up two things: "properties" and "attributes".

**Properties** are a notion introduced by the original OpenFX API. It is a generic way to attach information to various elements of the API, like effects, nodes, parameters, etc. In an Object Oriented mindset, it can be seen as "attributes" (hence the confusion, I guess) that can be get and set.

I would like to use the very same `ofxProperty.h` as [theirs](https://github.com/ofxa/openfx/blob/master/include/ofxProperty.h). They did not judge useful to introduce short properties, yet the API is widely used, so unless we have a particular use case that arises, I would not be in favor of adding it and pay the cost from breaking compatibility with the original OpenFX.

There is a lot to clean up in the current implementation of properties, I should add a few issues to the bug tracker to list them.

**Attributes** are extra mesh data attached either on points, vertices or faces. For instance, UVs are an attribute of type float2 attached to vertices, resulting in an extra buffer allocated in the mesh structure, besides point positions and topological information. My inspiration for this part of the API, that is defined in `ofxMeshEffect.h`, comes largely from Houdini's HAPI. In this API, attributes can be int, float, in64 or float64, as you can see [here](https://www.sidefx.com/docs/hengine/_h_a_p_i__attributes.html#HAPI_Attributes_SettingAttributeValues) with functions like `HAPI_SetAttributeFloatData()`, `HAPI_SetAttributeFloat64Data()`, etc.

Here again, since they did not bother introducing half floats (which would be float16) I trusted their experience and did not neither, but since this is part of the API we can still change, I am open to merging it.

Long story short, remove half properties and only keep half attributes for now!
Thanks.

PS: When you say it breaks existing plugins, did you compile them with the same version of ofx headers? And in which way does it break precisely?

**Comment by EitanSomething** [comment]
Custom normals in the MVert struct in blender uses shorts

**Comment by eliemichel** [comment]
All right, this is a fair use case for short attributes, which I agree on merging, but I still believe that short properties are not needed!

And same goes for parameters: there is not short param in original API, and I don't think we need some. ;)

**Comment by EitanSomething** [comment]
If we don't use efficient variables the modifiers will end up using a lot more memory.

**Comment by EitanSomething** [comment]
With the mirror plugin the mesh isn’t visible in object mode and it doesn’t mirror

**Comment by eliemichel** [comment]
What matters regarding memory is only attributes, because there is one value per geometric element (point, vertex, face) but for properties and parameters, which are around 10 per object, it's peanut.

**Comment by EitanSomething** [comment]
Fixed the problem. I had to rebuild the mirror ofx files

**Comment by EitanSomething** [comment]
I have to clean up the code and remove the parameters because all we care about is the attributes
